lib/views/pay/pay_view.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:clnapp/api/api.dart';
+import 'package:clnapp/components/buttons.dart';
+import 'package:clnapp/model/app_model/pay_invoice.dart';
+import 'package:clnapp/utils/app_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class PayView extends StatefulWidget {
+  final AppProvider provider;
+
+  const PayView({Key? key, required this.provider}) : super(key: key);
+  @override
+  State<PayView> createState() => _PayViewState();
+}
+
+class _PayViewState extends State<PayView> {
+  late String boltString;
+  late int amountMsat;
+  AppPayInvoice? paymentResponse;
+
+  Future<AppPayInvoice> payInvoice(String boltString, int amountMsat) async {
+    try {
+      final response = await widget.provider
+          .get<AppApi>()
+          .payInvoice(invoice: boltString, msat: amountMsat);
+      return response;
+    } catch (ex) {
+      AppPayInvoice error =
+          AppPayInvoice(payResponse: {"Error": ex.toString()});
+      return error;
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    amountMsat = -1;
+    boltString = "";
+    paymentResponse = null;
```

Comment:
I like the idea to init the state inside `initState` and not inside the constructor, but I was wondering if the init state trigger something that required that `amountMsat` will be initialized?

Reply:
Yeah will make `amountMsat` nullable